src/runtime_src/core/common/api/xrt_kernel.cpp

```diff
@@ -1217,6 +1217,24 @@ class run_impl
     }
   };
 
+  //Soft Kernel Arg setter
+  struct sk_arg_setter : arg_setter
+  {
+    sk_arg_setter(uint32_t* data)
+      : arg_setter(data)
+    {}
+
+    //Soft kernel regmap size and arguments are unknown; xclbin doesn't have this info
+    virtual void
+    set_arg_value(const uint8_t* args, const uint32_t size) //sizein bytes from 0x0
```

Comment:
How and where will this function be called, it doesn't match the signature of set_arg_value?

Reply:
See more info & code. For soft kernels args vector will be of size zero. For running kernel only full regmap setting will be allowed.